linera-chain/src/chain.rs

```diff
@@ -971,9 +978,12 @@ where
             messages.push(messages_out);
         }
 
-        // Finally, charge for the block fee, except if the chain is closed. Closed chains should
-        // always be able to reject incoming messages.
-        if !self.is_closed() {
+        // Finally, charge for the block fee, except if the chain is closed and the block
+        // was used to reject some incoming messages (and only that).
+        if !self.is_closed()
+            || block.incoming_messages.is_empty()
+            || !block.has_only_rejected_messages()
```

Comment:
Is the third line needed?
If it is closed, doesn't that imply that it has only rejected messages?

Reply:
After this PR, it could have been closed in the middle of the block.